lib/Promises/Channel.pm

```diff
@@ -95,6 +95,9 @@ has outbox =>
   is      => 'ro',
   default => sub { [] };
 
+has on_shutdown =>
+  is      => 'ro',
+  default => sub { deferred };
```

Comment:
In order to keep callers away from the deferred object itself, should this be stored in a more internal attribute, and a method added which returns its promise instead? For example:

```
has _on_shutdown =>
  is => 'ro',
  default => sub{ deferred };

has on_shutdown =>
  is => 'lazy';

sub _build_on_shutdown {
  my $self = shift;
  return $self->_on_shutdown->promise;
}
```